src/lib/util/Config.ts

```diff
@@ -122,10 +125,12 @@ export class Config {
 
 	private async upgradeConfig(): Promise<void> {
 		if (this.versionUpgradeManager.upgradeRequired(this.configContent.currentVersion)) {
-			// Start by creating a copy of the existing config, so we have it if we need it to create a backup file.
-			// Also determine where such a file would go.
+			// Start by cloning the existing config, so we have it if we need to create a back-up later.
+			// Also, determine whether a back-up should be created always or just for failed upgrades,
+			// and determine where such a back-up would go.
 			const existingConfig = deepCopy(this.configContent);
-			const backupFileName = `${CONFIG_PILOT_FILE}.${existingConfig.currentVersion || 'pre-2.7.0'}.bak`;
+			let shouldCreateBackup = this.versionUpgradeManager.versionNecessitatesBackup(this.configContent.currentVersion);
```

Comment:
Invocation of the new `VersionUpgradeManager` method.
If upgrading from a previous version (i.e. 2.x), we'll always save a back-up of the old config, instead of only doing so for failures.
Forced back-ups for v3.x are unnecessary since the pilot config is its own backup.